spec/unit/actions/ensure_data_product_spec.rb

```diff
@@ -0,0 +1,37 @@
+# encoding: UTF-8
+#
+# Copyright (c) 2010-2017 GoodData Corporation. All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+require 'gooddata/lcm/actions/ensure_data_product'
+require 'gooddata/lcm/lcm2'
+
+describe GoodData::LCM2::EnsureDataProduct do
+  context 'when data_product parameter is passed' do
+    let(:client) { double(:client) }
+    let(:domain) { double(:domain) }
+    let(:data_product) { GoodData::DataProduct.new({}) }
+    let(:gdc_logger) { double(:gdc_logger) }
+    let(:params) do
```

Comment:
Identical code found in collect_data_product_spec.rb:23, ensure_data_product_spec.rb:16 (mass = 76)